lib/internal/modules/esm/fetch_module.js

```diff
@@ -52,6 +52,9 @@ function HTTPSGet(url, opts) {
   });
   return https.get(url, {
     agent: HTTPSAgent,
+    headers: {
+      'User-Agent': `Node.js/${process.version}`,
+    },
```

Comment:
I'm -1 on always setting this. It should be optional and default turned off.

Reply:
How do other server-side runtimes behave?

Reply:
My inspiration for this was how Deno sets the User-Agent header when getting packages. I don't quite understand why so many people are against this -- many very popular packages and utilities do this. Curl sets a header. So does node-fetch. List goes on.

Reply:
“many people do x” isn’t ever a justification; many people do many ill-advised things.

The advantages i can see of setting it by default is easier debugging on the other end - the disadvantages i see is that a server could identify, track, and treat differently a request coming from node vs from other sources.

Reply:
> Deno sets the User-Agent header when getting packages. I don’t quite understand why so many people are against this – many very popular packages and utilities do this. Curl sets a header. So does node-fetch.

If most non-browser clients set a `User-Agent` header, then it’s a de facto standard and we should do the same. It doesn’t matter if some of us disagree with the wisdom of the practice; the point of adding APIs like `fetch` is to be more standards-compliant and similar to other clients. If users want network requests without `User-Agent`, they can just override the default or use some other API.